corehq/apps/app_manager/views/utils.py

```diff
@@ -167,7 +167,22 @@ def overwrite_app(app, master_build, report_map=None):
     ])
     master_json = master_build.to_json()
     app_json = app.to_json()
-    form_ids_by_xmlns = _get_form_ids_by_xmlns(app_json)  # do this before we change the source
+
+    # Corresponding forms in a master app and linked app need to have the same XMLNS but different unique ids,
+    # so the linked app needs to know if there are any new forms and, if so, assign those forms new unique ids.
+    # To do this lookup, get the XMLNSes from the the most recent versions of this app pulled from each master
+    # and compare those to the XMLNSes present in this app.
```

Comment:
has this been profiled for ICDS? Seems like it will be very slow

Reply:
I haven't profiled. It certainly could get slow, if any of the apps grow to have a lot of master apps and old one aren't deleted (which seems plausible). Added 54e68845c1c to reduce the amount of work this does, so it'll stop iterating once all XMLNSes have been found.